linkerd/app/core/src/proxy/server.rs

```diff
@@ -137,30 +164,32 @@ where
         make_http: H,
         h2_settings: H2Settings,
         drain: drain::Watch,
-    ) -> Self {
-        let connect = ForwardConnect(connect, PhantomData);
-        Self {
-            http: hyper::server::conn::Http::new(),
-            h2_settings,
-            transport_labels,
-            transport_metrics,
-            connect,
-            make_http,
-            drain,
-        }
+        skip_ports: Arc<IndexSet<u16>>,
+    ) -> detect::Accept<ProtocolDetect, Self> {
+        detect::Accept::new(
+            ProtocolDetect { skip_ports },
+            Self {
```

Comment:
This is really the whole crux of this PR.

Reply:
I found the return value of this to be a little surprising, because it returns an `detect::Accept` and then `server` is passed into `TlsAccept::new` (on inbound). The trait bound for `accept` in `TlsAccept::new` is that it implements `listen::Accept`, but `detect::Accept` only implements `Service`. Then I remembered the blanket `Accept` impl for all `Services`s that meet fairly simple bounds which is the case here.

All this to say I'm just writing this down now for my understanding, but I did not find it to be immediately obvious.